### PR TITLE
[#100] Chore replace catching logic with extension

### LINKED ISSUE #100: Update repo implementations to catch flow using extension
## Why

Currently catching flow is working but repeating codes. Create an extension to catch the flow and avoid boilerplate code. Replace the following with the extension:

```
. . .
} .catch { error ->
   emit(ResourceState.Error(error.message.orEmpty()))
}
```

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #100 

## What happened 👀

Currently catching flow is working but repeating codes. Create an extension to catch the flow and avoid boilerplate code. Replace the with the extension.

## Insight 📝

- Replace every `flow catch` with extension `catchError`
- Fix blank error text showing when there is a network error

## Proof Of Work 📹

Still catching the error:
https://user-images.githubusercontent.com/32578035/191212936-94727866-10d2-4b48-8a8c-4f8651a2b915.mp4

### CODE REVIEW

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt [hunk 1]
```diff
@@ -40,9 +41,7 @@ class HomeRepoImpl @Inject constructor(
             is ResourceState.Error -> emit(ResourceState.Error(apiResult.error))
             else -> emit(ResourceState.NetworkError)
         }
-    }.catch { error ->
-        emit(ResourceState.Error(error.message.orEmpty()))
-    }
+    }.catchError()
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt:44]: Why not replace the `flow {}` method with a method to catch errors entirely? [Something like this](https://github.com/nimblehq/android-templates/blob/5b611f3ca5d78e6363bd8693c955a346a589b6f6/template/data/src/main/java/co/nimblehq/template/data/extensions/ResponseMapping.kt#L20) 😉 What do you think? 🤔

Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt:44]: @ryan-conway I am catching in the `RepoImpl` using `safeApiCall` extension and when I try using [something like this](https://github.com/nimblehq/android-templates/blob/5b611f3ca5d78e6363bd8693c955a346a589b6f6/template/data/src/main/java/co/nimblehq/template/data/extensions/ResponseMapping.kt#L20), it doesn't make a huge difference for me. So I am planning to leave it like that.

Comment (ryan-conway) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt:44]: Understood, just know that we will need to add that extension method to _all_ flow calls 🤔

Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/home/HomeRepoImpl.kt:44]: @ryan-conway Yes, I did.

**Review by ryan-conway — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]
LGTM 🚀

**Pull request merged**